apps/server/src/modules/tool/uc/lti-tool.uc.ts

```diff
@@ -0,0 +1,59 @@
+import { LtiToolRepo } from '@shared/repo';
+import { LtiToolDO } from '@shared/domain/domainobject/ltitool.do';
+import { ICurrentUser, IFindOptions, Page, Permission, User } from '@shared/domain';
+import { AuthorizationService } from '@src/modules/authorization';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class LtiToolUc {
+	constructor(private readonly ltiToolRepo: LtiToolRepo, private readonly authorizationService: AuthorizationService) {}
+
+	async findLtiTool(
+		currentUser: ICurrentUser,
+		query: Partial<LtiToolDO>,
+		options: IFindOptions<LtiToolDO>
+	): Promise<Page<LtiToolDO>> {
+		const user: User = await this.authorizationService.getUserWithPermissions(currentUser.userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_VIEW]);
+
+		const tool: Promise<Page<LtiToolDO>> = this.ltiToolRepo.find(query, options);
```

Comment:
we need to prevent access without a valid query. teachers should only be able to either request only templates, or only those of a specific course. These seem to me like different usecases too, since the authorisation might differ (students probably need to see the tools of the course, but not the templates)

 For the course variant, changing the route to one including a course id would be advisable to make the api clearer, like /courses/:id/ltitools